### ISSUE TITLE
cascading.tuple.TupleException: unable to read from input identifier: 'unknown'

### ISSUE THREAD

**Issue by nikolovivan** [tresata/spark-scalding#1]
I've been trying to find my way around this issue, but I have no luck. I cannot even run the demo on my cluster. I follow the steps exactly as explained and I run my command as follows:

```
spark-submit --class com.twitter.scalding.Tool --master yarn-client /home/inikolov/demo.jar com.tresata.spark.scalding.demo.DemoJob --hdfs --input /user/inikolov/fruits.bsv --output /user/inikolov/test
```

I've tried prepending `hdfs://` to the paths as well, but the outcome is still the same.

Among the things I've tried was to run my custom code that reads SequenceFiles rather than just a text file, I've played with excluding dependencies, forcing dependency versions, etc. to what my cluster is running. Running in local mode also doesn't help.

Please find below some more info about my current set up:
1. Spark version: spark-1.1.1-bin-hadoop2.3.
2. Hadoop version: 2.3.0-cdh5.0.3.
3. In my sequence file example the files have been generated with **0.11.1-cdh5** version of scalding, so I've tried forcing this one for my job.

Here is a bit more of the stack trace I see:

```
15/01/26 16:03:04 ERROR TaskSetManager: Task 0 in stage 1.0 failed 4 times; aborting job
15/01/26 16:03:04 INFO YarnClientClusterScheduler: Removed TaskSet 1.0, whose tasks have all completed, from pool
15/01/26 16:03:04 INFO YarnClientClusterScheduler: Cancelling stage 1
15/01/26 16:03:04 INFO DAGScheduler: Failed to run runJob at CascadingRDD.scala:161
Exception in thread "main" java.lang.Throwable: If you know what exactly caused this error, please consider contributing to GitHub via following link.
https://github.com/twitter/scalding/wiki/Common-Exceptions-and-possible-reasons#orgapachesparksparkexception
at com.twitter.scalding.Tool$.main(Tool.scala:132)
at com.twitter.scalding.Tool.main(Tool.scala)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:606)
at org.apache.spark.deploy.SparkSubmit$.launch(SparkSubmit.scala:329)
at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:75)
at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
Caused by: org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 1.0 failed 4 times, most recent failure: Lost task 0.3 in stage 1.0 (TID 3, bl-hadoop-p19.prodsec.bl.visualdna.com): cascading.tuple.TupleException: unable to read from input identifier: 'unknown'
    cascading.tuple.TupleEntrySchemeIterator.hasNext(TupleEntrySchemeIterator.java:127)
    scala.collection.convert.Wrappers$JIteratorWrapper.hasNext(Wrappers.scala:41)
    scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:327)
    scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:327)
    scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:327)
    scala.collection.Iterator$$anon$11.hasNext(Iterator.scala:327)
    org.apache.spark.util.collection.ExternalAppendOnlyMap.insertAll(ExternalAppendOnlyMap.scala:144)
    org.apache.spark.Aggregator.combineValuesByKey(Aggregator.scala:58)
    org.apache.spark.shuffle.hash.HashShuffleWriter.write(HashShuffleWriter.scala:55)
    org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:68)
    org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:41)
    org.apache.spark.scheduler.Task.run(Task.scala:54)
    org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:178)
    java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
    java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
    java.lang.Thread.run(Thread.java:745)
Driver stacktrace:
at org.apache.spark.scheduler.DAGScheduler.org$apache$spark$scheduler$DAGScheduler$$failJobAndIndependentStages(DAGScheduler.scala:1185)
at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:1174)
at org.apache.spark.scheduler.DAGScheduler$$anonfun$abortStage$1.apply(DAGScheduler.scala:1173)
at scala.collection.mutable.ResizableArray$class.foreach(ResizableArray.scala:59)
at scala.collection.mutable.ArrayBuffer.foreach(ArrayBuffer.scala:47)
at org.apache.spark.scheduler.DAGScheduler.abortStage(DAGScheduler.scala:1173)
at org.apache.spark.scheduler.DAGScheduler$$anonfun$handleTaskSetFailed$1.apply(DAGScheduler.scala:688)
at org.apache.spark.scheduler.DAGScheduler$$anonfun$handleTaskSetFailed$1.apply(DAGScheduler.scala:688)
at scala.Option.foreach(Option.scala:236)
at org.apache.spark.scheduler.DAGScheduler.handleTaskSetFailed(DAGScheduler.scala:688)
at org.apache.spark.scheduler.DAGSchedulerEventProcessActor$$anonfun$receive$2.applyOrElse(DAGScheduler.scala:1391)
at akka.actor.ActorCell.receiveMessage(ActorCell.scala:498)
at akka.actor.ActorCell.invoke(ActorCell.scala:456)
at akka.dispatch.Mailbox.processMailbox(Mailbox.scala:237)
at akka.dispatch.Mailbox.run(Mailbox.scala:219)
at akka.dispatch.ForkJoinExecutorConfigurator$AkkaForkJoinTask.exec(AbstractDispatcher.scala:386)
at scala.concurrent.forkjoin.ForkJoinTask.doExec(ForkJoinTask.java:260)
at scala.concurrent.forkjoin.ForkJoinPool$WorkQueue.runTask(ForkJoinPool.java:1339)
at scala.concurrent.forkjoin.ForkJoinPool.runWorker(ForkJoinPool.java:1979)
at scala.concurrent.forkjoin.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:107)
```

I've looked around the source code, I've tried changing things, but nothing has changed for me. I would be really thankful if you could give me any pointers at what else I could look into.

**Comment by koertkuipers** [comment]
hey,
thanks for reporting this.

the exception you are seeing in the task (cascading.tuple.TupleException: unable to read from input identifier: 'unknown') is thrown by TupleEntrySchemeIterator if another error occurred while getting the next tuple.

can you provide me with the full stacktrace of the exception in the task? or better, the full logs of the failed task?
you can email them to koert at tresata.com

thanks

**Comment by nikolovivan** [comment]
Hey Koert,

Thanks for the really quick reply and sorry for not getting back to you earlier. I just managed to get proper access to my computer and I decided to give it another few tries (by the way, that was the full log I could see anyway).

I completely recompiled the library by replacing the current version of Scalding and Spark with the ones I am using (with my custom job I used the `force()` sbt directive). Not sure why I didn't try this earlier, but finally the demo you provided worked, so I expect things would also work for the actual thing I'm trying to do.

Great job on the library and thanks again!
